### ISSUE TITLE
Switching languages doesn't take effect on some menu elements

### ISSUE THREAD

**Issue by Atmostone** [apache/superset#29882]
### Bug description

Problem appears when I set default language not to English.
If I add next lines to config file:

`BABEL_DEFAULT_LOCALE = 'ru'

LANGUAGES = {
'ru': {'flag': 'ru', 'name': 'Русский'},
'en': {'flag': 'us', 'name': 'English'},
}`

Then English version breaks (most of menu links are still on Russian).

For example RU version:
![image](https://github.com/user-attachments/assets/7238e170-433f-4326-8668-e9e7b696994d)

And broken EN version:
![image](https://github.com/user-attachments/assets/fd44cb54-a647-420e-9824-8e688c1f9db0)

### How to reproduce the bug

1. Add next lines to config file:

`BABEL_DEFAULT_LOCALE = 'ru'

LANGUAGES = {
'ru': {'flag': 'ru', 'name': 'Русский'},
'en': {'flag': 'us', 'name': 'English'},
}`

2. Switch language to English
3. Menu is still on Russian

### Screenshots/recordings

![image](https://github.com/user-attachments/assets/68b54259-5546-4381-920c-19b39ad22ab9)
![image](https://github.com/user-attachments/assets/b3a6e558-4766-480a-8f91-0d2893c2477a)

### Superset version

master / latest-dev

### Python version

3.11

### Node version

18 or greater

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by Atmostone** [comment]
@dosu I already deleted LANGUAGES = {}, no effect. Menu elements are still written on language set by BABEL_DEFAULT_LOCALE. But if BABEL_DEFAULT_LOCALE set to "en", then there is no any problems.

**Comment by Atmostone** [comment]
@dosu Changing `locale` in `menu_data` didn't do anything useful,  config.py is already correct, problem is somewhere else

**Comment by goldjee** [comment]
We've stumbled upon that in our environment too. As a workaround, we have switched the default back to English, still providing users an option to select Russian.

**Comment by Atmostone** [comment]
Does anyone know how can it be solved?
Plus, tag for this issue should be "i18n", not "i18n:russian", because this problem seems to appear in all languages

**Comment by Atmostone** [comment]
All strings that are not translated are located in superset/initialization/__init__.py file.
So it seems that their locale is set when superset starts up and then changing language doesn't affect it.
Not sure how can it be solved though

**Comment by Atmostone** [comment]
I found how to fix this bug, but I can't push my fix branch to superset repository (_unable to access 'https://github.com/apache/superset.git/': The requested URL returned error: 403_). Does anyone know how to push it?

**Comment by goldjee** [comment]
It's normal that you can't push your changes to this repository directly. You need to fork apache/superset, create a new branch in your fork, commit your changes to it and create a pull request.

You can find some useful information on contributing here: https://superset.apache.org/docs/contributing/

**Comment by rusackas** [comment]
Yep, and please ping me on the pull request, @Atmostone, I'll be happy to help review it.

**Comment by Iloer** [comment]
@Atmostone could you describe the essence of the solution. thank you!

**Comment by Atmostone** [comment]
Solution is to this bug is to change all `gettext` in _superset/initialization/__init__.py_ to `lazy_gettext`.
Then everything works as it should and languages switch without problems

**Comment by rusackas** [comment]
Is anyone willing to open a PR to that effect if the above is indeed the solution?

**Comment by Ronkiro** [comment]
Also found this error, #34049 seems to fix it
